File: app/src/main/java/com/github/colinjeremie/willyoufindit/utils/FilterRadioListTask.java
package com.github.colinjeremie.willyoufindit.utils;

import android.os.AsyncTask;

import com.deezer.sdk.model.Radio;

import java.util.ArrayList;
import java.util.List;

/**
 * * WillYouFindIt
 * Created by jerem_000 on 4/23/2016.
 */
public class FilterRadioListTask extends AsyncTask<String, Void, List<Radio>> {
    private final OnListFilteredListener mCallBack;
    private final List<Radio> mOriginalValues;

    public FilterRadioListTask(List<Radio> pValues, OnListFilteredListener pCallback){
        this.mOriginalValues = pValues;
        this.mCallBack = pCallback;
    }

    @Override
    protected List<Radio> doInBackground(String ... params) {
        if (params.length > 0){
            String search = NormalizeStr.normalizeIt(params[0]).toLowerCase();
            List<Radio> valuesFiltered = new ArrayList<>();

            for (Radio radio : mOriginalValues){
                String title = NormalizeStr.normalizeIt(radio.getTitle()).toLowerCase();

                if (title.contains(search)){
                    valuesFiltered.add(radio);
                }
            }
            return valuesFiltered;
        }
        return mOriginalValues;
    }

    @Override
    protected void onPostExecute(List<Radio> radios) {
        super.onPostExecute(radios);
        if (mCallBack != null){
            mCallBack.onListFiltered(radios);
        }
    }

    public interface OnListFilteredListener{
        void onListFiltered(List<Radio> pList);
    }
}

File: app/src/main/java/com/github/colinjeremie/willyoufindit/utils/FilterRadioListTask.kt
package com.github.colinjeremie.willyoufindit.utils

import android.os.AsyncTask

import com.deezer.sdk.model.Radio

import java.util.ArrayList

class FilterRadioListTask(private val originalValues: List<Radio>,
                          private val callBack: OnListFilteredListener?) : AsyncTask<String, Void, MutableList<Radio>>() {

    override fun doInBackground(vararg params: String): MutableList<Radio> {
        if (params.isNotEmpty()) {
            val search = params[0].normalize().toLowerCase()
            val valuesFiltered = mutableListOf<Radio>()

            for (radio in originalValues) {
                val title = radio.title.normalize().toLowerCase()

                if (title.contains(search)) {
                    valuesFiltered.add(radio)
                }
            }
            return valuesFiltered
        }
        return originalValues.toMutableList()
    }

    override fun onPostExecute(radios: MutableList<Radio>) {
        super.onPostExecute(radios)
        callBack?.onListFiltered(radios)
    }

    interface OnListFilteredListener {
        fun onListFiltered(list: MutableList<Radio>)
    }
}
